apps/email/js/sync.js

```diff
@@ -321,7 +320,7 @@ define(function(require) {
               });
             }
 
-            body = newFromNames.join(mozL10n.get('senders-separation-sign'));
+            body = newFromNames.join('senders-separation-sign');
```

Comment:
So, here you need to use an async version of `mozL10n.get` called `mozL10n.formatValue` - it returns a promise.

```javascript
navigator.mozL10n.formatValue('senders-separation-sign').then(separator => {
  body = newFromNames.join(separator);
});
```

and then the rest of the logic has to stay inside the Promise callback.